### ISSUE TITLE
tree: lazy loading

### ISSUE THREAD

**Issue by oliver-sanders** [cylc/cylc-ui#1228]
Currently the tree view creates components for the whole thing on load.

This is slow and unnecessary as we don't need the whole tree to be loaded, just the nodes which are expanded.

We should be able to do something like this:

```
<tree-item
  v-if="isLoaded"
  v-show="isExpanded"
/>
```

To avoid recursively creating the tree before it is needed, where `isLoaded` defaults to `isExpanded` and is set to true if `isExpanded` becomes true at any point.

This will make n=1 views more efficient. For n=cycle views we will also need the ability to set `isLoaded` to false when we collapse cycle point nodes in order to delete the component tree for that cycle for efficiency reasons (and also because we will be purging the underlying data).

**Pull requests welcome!**

**Comment by MetRonnie** [comment]
I think we need to be careful to ensure filtering by task ID would still work, even if the task has been collapsed under a family

![image](https://user-images.githubusercontent.com/61982285/219025564-cd0da12f-1deb-4d21-80a4-8a753fffef17.png)

**Comment by oliver-sanders** [comment]
Good point, the filtering would need to be performed on the "virtual" tree rather than within the components themselves which does complicate it somewhat.
